tests/i18n/test_extraction.py

```diff
@@ -868,6 +868,32 @@ def test_add_location_gettext_version_check(self):
             )
 
 
+class ObsoleteTranslationTests(ExtractorTests):
+    work_subdir = "obsolete_translations"
+
+    PO_FILE = "locale/%s/LC_MESSAGES/django.po"
+
+    def test_no_obsolete(self):
+        management.call_command(
+            "makemessages", locale=[LOCALE], verbosity=0, no_obsolete=True
+        )
+        self.assertTrue(os.path.exists(self.PO_FILE % LOCALE))
+        with open(self.PO_FILE % LOCALE) as fp:
+            po_contents = fp.read()
+            self.assertNotIn(
+                '#~ msgid "This is another translatable string."', po_contents
+            )
+            self.assertNotIn(
+                '#~ msgstr "This is another translated string."', po_contents
+            )
```

Comment:
The second assertion is unnecessary:
```suggestion
            self.self.assertNotMsgId("This is another translatable string.", po_contents)
```

Reply:
Reverted, we cannot use `assertNotMsgId()` because obsolete strings start with `#~`.